### PR TITLE
Add working example of image transformation

### PR DESCRIPTION
**Pull request by wardVD**
Add image TFRecords + IPython notebook to illustrate how to transform images with Transform component.

### CODE REVIEW

File: chapters/data_processing/transform_images.ipynb [hunk 1]
```diff
@@ -0,0 +1,4003 @@
...
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "import tensorflow as tf\n",
+    "import IPython.display as display\n",
+    " \n",
+    "raw_image_dataset = tf.data.TFRecordDataset('./data/tfrecords_small_tfrecord-00000-of-00001')\n",
```
Comment (hanneshapke) [inline chapters/data_processing/transform_images.ipynb:88]: Do you mind renaming the file to `tfrecords_small.tfrecord-00000-of-00001` to follow the naming convention from Google tutorials (https://www.tensorflow.org/tutorials/load_data/tfrecord#tfrecords_format_details)?

File: chapters/data_processing/transform_images.ipynb [hunk 2]
```diff
@@ -0,0 +1,4003 @@
...
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "example_gen = ImportExampleGen(input_base='./data',\n",
+    "                               output_config=output)\n",
+    " \n",
+    "context.run(example_gen, enable_cache=False)"
```
Comment (hanneshapke) [inline chapters/data_processing/transform_images.ipynb:655]: Is there a specific reason why you need to disable the cache here? If so, please leave a comment.

File: chapters/data_processing/transform_images.ipynb [hunk 3]
```diff
@@ -0,0 +1,4003 @@
...
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "statistics_gen = StatisticsGen(examples = example_gen.outputs['examples'])\n",
+    "context.run(statistics_gen, enable_cache=False)"
```
Comment (hanneshapke) [inline chapters/data_processing/transform_images.ipynb:1313]: Pls see earlier comment.

File: chapters/data_processing/transform_images.ipynb [hunk 4]
```diff
@@ -0,0 +1,4003 @@
...
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "schema_gen = SchemaGen(\n",
+    "    statistics=statistics_gen.outputs['statistics'],\n",
+    "    infer_feature_shape=True)\n",
+    "context.run(schema_gen, enable_cache=False)"
```
Comment (hanneshapke) [inline chapters/data_processing/transform_images.ipynb:2074]: Same here.

**Review by hanneshapke — CHANGES_REQUESTED** [review]
Hi @wardVD Thank you for your PR. Do you mind adding an additional cell after the transform step, reading the transformed_examples and displaying one image to show that they are b/w and 300x300? I had to call the `map_fn` with the output of a `lambda` call as shown in the Issue. Would be great to confirm that the batch isn't flattened when you call `tf.reshape(..., shape=[])`. Thank you!

File: chapters/data_processing/transform_images.ipynb (hunk 1 above)
Comment (wardVD) [reply chapters/data_processing/transform_images.ipynb:88]: Ok

File: chapters/data_processing/transform_images.ipynb (hunk 2 above)
Comment (wardVD) [reply chapters/data_processing/transform_images.ipynb:655]: Not really. Still experimenting :-)

File: chapters/data_processing/transform_images.ipynb (hunk 3 above)
Comment (wardVD) [reply chapters/data_processing/transform_images.ipynb:1313]: See above

File: chapters/data_processing/transform_images.ipynb (hunk 4 above)
Comment (wardVD) [reply chapters/data_processing/transform_images.ipynb:2074]: See above

**Comment by wardVD** [comment]
> Hi @wardVD Thank you for your PR. Do you mind adding an additional cell after the transform step, reading the transformed_examples and displaying one image to show that they are b/w and 300x300? I had to call the `map_fn` with the output of a `lambda` call as shown in the Issue. Would be great to confirm that the batch isn't flattened when you call `tf.reshape(..., shape=[])`. Thank you!

I didn't see a difference in either way. I've included both ways, one is commented out. They give me the same result.

**Comment by hanneshapke** [comment]
Hi @wardVD,
Wow, I didn't think Github would close the PR when I renamed the `master` branch. Sorry about that!  Could you please reopen the PR if possible. I am very sorry about the mistake.

**Pull request closed without merging**